pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -97,18 +97,22 @@ func (c *controller) processAddOrUpdateLC(logicalCluster *lcv1alpha1.LogicalClus
 		go provider.Create(c.context, logicalCluster.Name, pclient.Options{})
 		return nil
 	}
-	// case lcv1alpha1.LogicalClusterPhaseInitializing:
-	// A managed cluster is being created by the provider. Need to wait for
-	// the cluster to be created at which point discovery will change the
-	// state to READY and update the cluster config.
-	//
-	// case lcv1alpha1.LogicalClusterPhaseReady:
-	// The cluster is ready - we have nothing to do but celebrate!
-	// Likely we got here after the provider finished creating a managed
-	// cluster and dicovery moved it into the Ready state.
-	//
-	// If a logical cluster has been created for an unmanaged physical
-	// cluster, then wait for discovery to move the phase to Ready.
+
+	// The code from here deals with the case of provider deleted cluster events that might be missed
+	_, err = provider.Get(c.context, logicalCluster.Name)
```

Comment:
This is done on each event? What is the performance hit?

Reply:
General question - how frequent are LogicalCluster events? I would have thought that they occur only on cluster changes (which is infrequent) and on startup.  And what's the expected number of clusters?

Reply:
This will be taken from a cache I'm building